sampler.go

```diff
@@ -328,6 +328,11 @@ func (s *adaptiveSampler) Close() {
 }
 
 func (s *adaptiveSampler) Equal(other Sampler) bool {
+	// NB The Equal() function is overly expensive for adaptiveSampler since it's composed of multiple
+	// samplers which all need to be initialized before this function can be called for a comparison.
+	// Therefore, adaptiveSampler uses the update() function to only alter the samplers that need
+	// changing. Hence this function always returns false so that the update function can be called.
+	// Once the Equal() function is removed from the Sampler API, this will no longer be needed.
```

Comment:
I think similar comment can be added to remote sampler's equals method. It's never actually called by anyone.